test/partiql-tests-runner/src/test/kotlin/org/partiql/runner/executor/EvalExecutor.kt

```diff
@@ -84,7 +84,7 @@ class EvalExecutor(
         if (v1.isNull && v2.isNull) {
             return true
         }
-        if (v1 == v2) {
+        if (comparator.compare(v1, v2) == 0) {
```

Comment:
NULL/MISSING according to your other comment would cause problems here, right?

Reply:
Yes.

We should not use the comparator here, we should have the equals or hash code method implemented in the value class. 

But the comparator is at the moment the only comparator, specifically when it comes to bag comparison. 

I think we can leave this line here but add a TODO.

Reply:
Is there a specific scenario where this was failing? I can see that all of the value impls are data classes with the exception of collections and structs.